src/common/datetime.cpp

```diff
@@ -541,6 +541,52 @@ bool wxDateTime::IsLeapYear(int year, wxDateTime::Calendar cal)
     }
 }
 
+#ifdef _WIN32
+#include <wx/msw/registry.h>
+
+/* static */
+wxDateTime::WeekDay wxDateTime::GetFirstWeekDay(bool *success) {
+    wxRegKey key("HKEY_CURRENT_USER\\Control Panel\\International");
+    wxString val;
+
+    if (key.Exists() && key.HasValue("iFirstDayOfWeek")) {
+        key.QueryValue("iFirstDayOfWeek", val);
+        if (success)
+             *success = true;
+        return wxDateTime::WeekDay((wxAtoi(val) + 1) % 7);
+	} else {
+        if (success)
+             *success = false;
+        return wxDateTime::Sun;
+	}
+}
+
+#elif defined(__APPLE__)
+// implementation in utils_base.mm
+#elif defined(__linux__)
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+#include <langinfo.h>
+
+/* static */
+wxDateTime::WeekDay wxDateTime::GetFirstWeekDay(bool *success) {
+    if (success)
+        *success = true;
+    return wxDateTime::WeekDay((*nl_langinfo(_NL_TIME_FIRST_WEEKDAY) - 1) % 7);
```

Comment:
OTOH it might make sense to test for `_NL_TIME_FIRST_WEEKDAY` availability (using `#ifdef`). And we already have `HAVE_LANGINFO_H` for `nl_langinfo()` itself.

Like this, this code could work for non-Linux systems too if they use glibc (or implement this extension).